snapcraft/meta/appstream.py

```diff
@@ -102,7 +104,17 @@ def extract(relpath: str, *, workdir: str) -> Optional[ExtractedMetadata]:
     version = _get_latest_release_from_nodes(dom.findall("releases/release"))
     project_license = _get_value_from_xml_element(dom, "project_license")
     update_contact = _get_value_from_xml_element(dom, "update_contact")
-    contact = [update_contact] if update_contact else None
+    contact = None
+    if update_contact:
+        if urlparse(update_contact).scheme in ["http", "https"]:
+            contact = [update_contact]
+        else:
+            try:
+                EmailStr.validate(update_contact)
+            except:
+                pass
+            else:
+                contact = [update_contact]
```

Comment:
Something like this might be a bit more manageable (no blank exception catching either)
```suggestion
    if update_contact and urlparse(update_contact).scheme in ["http", "https"]:
        contact = [update_contact]
    else:
        try:
            EmailStr.validate(update_contact)
            contact = [update_contact]
        except pydantic.ValidationError as err:
            emit.debug(f"Invalid email {update_contact!r} set: {err!s}")
```

Reply:
This one is a bit problematic as what the appstream has passed might not be an email and a malformed website url, like eg "http:/testcom", I have found another better library/tool to fix this. But, this would create a new addition to the `requirements.txt`. 

https://pypi.org/project/validators/

And I think we can emit a general error that `invalid url {update_contact}`

Regarding test, yes, I also thought about it, and I think it should have its own unit test. Although, I did add one such case here already

[L528](https://github.com/canonical/snapcraft/pull/4888/files#diff-5eaf20f392bec62c4f476a461c87baf38d7f9710774bf720459bd1ea2808e408R528)
[L553](https://github.com/canonical/snapcraft/pull/4888/files#diff-5eaf20f392bec62c4f476a461c87baf38d7f9710774bf720459bd1ea2808e408R553)

But, I would definitely want to add one such test.